### PR TITLE
FEAT: Add covalent indexer in outport driver

### PR DESCRIPTION
**Pull request by mariusmihaic**
_**Note:**_ This PR **should not** be merged (although it will most likely not build/pass anyway) until a proper release/development from **covalent-indexer** repo is ready

- Created and subscribed covalent indexer in outport driver. By default it is not enabled. It is supposed to be enabled only for covalent's observer nodes
- Added unit test to subscribe covalent indexer
- Added some unit tests for each driver subscription. However, there is no test to successfully subscribe an elastic indexer

### CODE REVIEW

File: cmd/node/config/external.toml [hunk 1]
```diff
@@ -32,3 +32,7 @@
 
     # Password is used to authorize an observer to push event data
     Password = ""
+
+[CovalentConnector]
+    Enabled = true
+    URL = "http://localhost:1234"
```
Comment (iulianpascalau) [inline cmd/node/config/external.toml:38]: add empty line

File: config/externalConfig.go [hunk 2]
```diff
@@ -25,3 +26,8 @@ type EventNotifierConfig struct {
 	Username         string
 	Password         string
 }
+
+type CovalentConfig struct {
```
Comment (iulianpascalau) [inline config/externalConfig.go:30]: add comments on exported items

File: outport/factory/outportFactory.go [hunk 3]
```diff
@@ -39,9 +41,30 @@ func createAndSubscribeDrivers(outport outport.OutportHandler, args *OutportFact
 		return err
 	}
 
+	err = createAndSubscribeCovalentDriverIfNeed(outport, args.CovalentIndexerFactoryArgs)
```
Comment (iulianpascalau) [inline outport/factory/outportFactory.go:44]: ...IfNeeded *

File: outport/factory/outportFactory.go [hunk 4]
```diff
@@ -39,9 +41,30 @@ func createAndSubscribeDrivers(outport outport.OutportHandler, args *OutportFact
...
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
+func createAndSubscribeCovalentDriverIfNeed(
```
Comment (iulianpascalau) [inline outport/factory/outportFactory.go:52]: unit tests?

File: go.mod [hunk 5]
```diff
@@ -7,6 +7,7 @@ require (
 	github.com/ElrondNetwork/arwen-wasm-vm/v1_3 v1.3.30
 	github.com/ElrondNetwork/arwen-wasm-vm/v1_4 v1.4.8
 	github.com/ElrondNetwork/concurrent-map v0.1.3
+	github.com/ElrondNetwork/covalent-indexer-go v0.0.0-20210830114936-46b978c21f79
```
Comment (iulianpascalau) [inline go.mod:10]: proper release?

File: cmd/node/config/external.toml (hunk 1 above)
Comment (mariusmihaic) [reply cmd/node/config/external.toml]: Added empty line and some comments here

File: config/externalConfig.go (hunk 2 above)
Comment (mariusmihaic) [reply config/externalConfig.go:30]: Added consistent _(with previous structs comms)_ comment here

File: outport/factory/outportFactory.go (hunk 3 above)
Comment (mariusmihaic) [reply outport/factory/outportFactory.go:44]: Renamed it

File: outport/factory/outportFactory.go (hunk 4 above)
Comment (mariusmihaic) [reply outport/factory/outportFactory.go:52]: I've added:
- A unit test to subscribe covalent indexer 
- Some unit tests for each driver subscription, which also increased code coverage with ~40%. (currently ~98%)

However, I did create a test to successfully subscribe an elastic indexer

File: go.mod (hunk 5 above)
Comment (mariusmihaic) [reply go.mod]: Yes, this shall be changed when a proper release/development is ready. I've only added it to test it locally

**Review by AdoAdoAdo — APPROVED** [review]

File: go.mod [hunk 6]
```diff
@@ -60,3 +60,5 @@ replace github.com/ElrondNetwork/arwen-wasm-vm/v1_2 v1.2.30 => github.com/Elrond
 replace github.com/ElrondNetwork/arwen-wasm-vm/v1_3 v1.3.30 => github.com/ElrondNetwork/arwen-wasm-vm v1.3.30
 
 replace github.com/ElrondNetwork/arwen-wasm-vm/v1_4 v1.4.8 => github.com/ElrondNetwork/arwen-wasm-vm v1.4.8
+
+replace github.com/ElrondNetwork/covalent-indexer-go => /home/elrond/go/src/github.com/ElrondNetwork/covalent-indexer-go
```
Comment (iulianpascalau) [inline go.mod:64]: proper release + remove this replace + go mod tidy

File: outport/factory/outportFactory_test.go [hunk 7]
```diff
@@ -1,51 +1,124 @@
...
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			_, err := CreateOutport(tt.argsFunc())
+			_, err := factory.CreateOutport(tt.argsFunc())
 			require.True(t, errors.Is(err, tt.exError))
 		})
 	}
 }
+
+func TestCreateOutport_EnabledDrivers_NilMockArgs_ExpectErrorSubscribingDrivers(t *testing.T) {
```
Comment (iulianpascalau) [inline outport/factory/outportFactory_test.go:65]: TestCreateOutport_EnabledDriversNilMockArgsExpectErrorSubscribingDrivers

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: outport/factory/outportFactory_test.go (hunk 7 above)
Comment (mariusmihaic) [reply outport/factory/outportFactory_test.go:65]: Renamed it

File: go.mod (hunk 6 above)
Comment (mariusmihaic) [reply go.mod]: Added release + removed it !

**Review by iulianpascalau — DISMISSED** [review]

**Review by iulianpascalau — DISMISSED** [review]

**Review by AdoAdoAdo — DISMISSED** [review]

**Review by gabi-vuls — DISMISSED** [review]
System test passed.
image

**Review by AdoAdoAdo — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]

**Pull request merged**